pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -438,6 +498,107 @@ class TNCudaBase : public TensornetBase<PrecisionT, Derived> {
                       projected_mode_values, numHyperSamples);
     }
 
+  protected:
+    /**
+     * @brief Get a vector of pointers to tensor data of each site.
+     *
+     * @return std::vector<CFP_t *>
+     */
+    [[nodiscard]] auto getTensorsOutDataPtr() -> std::vector<CFP_t *> {
+        std::vector<CFP_t *> tensorsOutDataPtr(BaseType::getNumQubits());
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            tensorsOutDataPtr[i] = tensors_out_[i].getDataBuffer().getData();
+        }
+        return tensorsOutDataPtr;
+    }
+
+    /**
+     * @brief Get a vector of pointers to extents of each site.
+     *
+     * @return std::vector<int64_t const *> Note int64_t const* is
+     * required by cutensornet backend.
+     */
+    [[nodiscard]] auto getSitesExtentsPtr() -> std::vector<int64_t const *> {
+        std::vector<int64_t const *> sitesExtentsPtr_int64(
+            BaseType::getNumQubits());
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            sitesExtentsPtr_int64[i] = sitesExtents_int64_[i].data();
+        }
+        return sitesExtentsPtr_int64;
+    }
+
+    /**
+     * @brief Dummy tensor operator update to allow multiple calls of
+     * appendMPSFinalize. This is a workaround to avoid the issue of the
+     * cutensornet library not allowing multiple calls of appendMPSFinalize.
+     *
+     * This function either appends a new `Identity` gate to the graph when the
+     * gate cache is empty or update the existing gate operator by itself.
+     */
+    void dummy_tensor_update() {
+        if (identiy_gate_ids_.empty()) {
+            applyOperation("Identity", {0}, false);
+        }
+
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetStateUpdateTensorOperator(
+            /* const cutensornetHandle_t */ BaseType::getTNCudaHandle(),
+            /* cutensornetState_t */ BaseType::getQuantumState(),
+            /* int64_t tensorId*/
+            static_cast<int64_t>(identiy_gate_ids_.front()),
+            /* void* */
+            static_cast<void *>(
+                gate_cache_->get_gate_device_ptr(identiy_gate_ids_.front())),
+            /* int32_t unitary*/ 1));
```

Comment:
is this description correct? shouldn't it be something like compute tensor state?

Reply:
the description is correct since it's factoring the final state into MPS representations.